platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -245,291 +518,180 @@ message RoundRosterPair {
-  <p>
-  @param genesisRoster the Genesis Roster for the network.
+/**
+ * Read-only implementation for accessing rosters states.
  */
-PlatformBuilder withGenesisRoster(@NonNull final Roster genesisRoster);
-```
-
-The logic to determine whether to invoke this method could be determined by some mechanism Services deems appropriate.
-
-#### Override Roster
+public interface ReadableRosterStore {
+    /**
+     * Gets the candidate roster if found in state or null otherwise.
+     * Note that state commits are buffered,
+     * so it is possible that a recently stored candidate roster is still in the batched changes and not yet committed.
+     * Therefore, callers of this API must bear in mind that an immediate call after storing a candidate roster may return null.
+     *
+     * @return the candidate roster
+     */
+    @Nullable
+    Roster getCandidateRoster();
 
-An `Override Roster` is an optional Roster that DevOps may provide for the explicit purpose of starting a network using
-a specific state, with a specific set of nodes, such as during a Network Transplant Process. A common example of
-this is Mainnet State being transplanted into a testing network.
+    /**
+     * Gets the active roster.
+     * Returns the active roster iff:
+     *      the roster state singleton is not null
+     *      the list of round roster pairs is not empty
+     *      the first round roster pair exists
+     *      the active roster hash is present in the roster map
+     * otherwise returns null.
+     * @return the active roster
+     */
+    @Nullable
+    Roster getActiveRoster();
 
-A quick note on PCES. The software version is used to determine which roster (active or previous) should be used to
-validate the signatures of events in the PCES. If the PCES event version is less than the current software version,
-it indicates that the event was signed with the old roster, and hence, the old roster is used for validation. If the
-PCES event version matches the current software version, the active roster is used. This behavior will be
-maintained until Birth Rounds or Dynamic Address Book is implemented.
+    /**
+     * Gets the roster history.
+     * Returns the active roster history iff:
+     *      the roster state singleton is not null
+     *      the list of round roster pairs is not empty
+     *      the active roster hashes are present in the roster map
+     * otherwise returns null.
+     * @return the active rosters
+     */
+    @Nullable
+    List<RoundRosterPair> getRosterHistory();
 
-Therefore, Provided an existing State, an `Override Roster` and a network upgrade (which could be a simple
-config-only upgrade if a software version upgrade isn't required), the platform will adopt the provided Override Roster.
+    /**
+     * Get the roster based on roster hash
+     *
+     * @param rosterHash The roster hash
+     * @return The roster.
```

Comment:
I assume this method returns `null` if the roster is not found. We should add this in the JavaDoc.

Reply:
Adding documentation that it returns null.  I won't resolve this comment until I create a PR that corrects the javadoc in the code.